openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityRepository.java

```diff
@@ -582,6 +588,53 @@ public PutResponse<T> addFollower(String updatedBy, UUID entityId, UUID userId)
     return new PutResponse<>(Status.OK, changeEvent, RestUtil.ENTITY_FIELDS_CHANGED);
   }
 
+  @Transaction
+  public PutResponse<T> updateVote(String updatedBy, UUID entityId, VoteRequest request) throws IOException {
+    // Get entity
+    T originalEntity = dao.findEntityById(entityId);
+
+    // Validate User
+    User user = daoCollection.userDAO().findEntityByName(updatedBy);
+    UUID userId = user.getId();
+    if (Boolean.TRUE.equals(user.getDeleted())) {
+      throw new IllegalArgumentException(CatalogExceptionMessage.deletedUser(userId));
+    }
+
+    ChangeDescription change = new ChangeDescription().withPreviousVersion(originalEntity.getVersion());
+
+    // Add or Delete relationship
+    if (request.getUpdatedVoteType() == VoteRequest.VoteType.UN_VOTED) {
```

Comment:
are we not going to support the DownVoted?

Reply:
you need to have 3 conditions to work here
1. User upVotes
2. User removes their upVote
3. User downVotes or changes their upVote from downVote by clicking the downVote button
4. user removes their downVote

At any given point user either not voted, or upVoted or downVoted there can't be in 2 states

Reply:
hi @harshach, we maintain a relationship for UP_VOTED and DOWN_VOTED both, but the relationship for the user is not there in case he has not voted, so if UI sends UN_VOTED we remove the relationship, or else if it sends UP_VOTED or DOWN_VOTED, in that case, we maintain USER -> VOTED -> DATA_ASSET and store the voting type in schema that is UPVOTED or DOWNVOTED